plugins/inputs/ceph/ceph.go

```diff
@@ -85,9 +103,23 @@ func (c *Ceph) gatherAdminSocketStats(acc telegraf.Accumulator) error {
 			continue
 		}
 		for tag, metrics := range data {
-			acc.AddFields(measurement,
-				metrics,
-				map[string]string{"type": s.sockType, "id": s.sockID, "collection": tag})
+			for name, metric := range metrics {
+				var valueType telegraf.ValueType
+				var ok bool
+				if valueType, ok = c.SchemaMaps[*s][tag][name]; !ok {
+					acc.AddGauge(measurement,
+						map[string]interface{}{name: metric},
+						map[string]string{"type": s.sockType, "id": s.sockID, "collection": tag})
+				}
+				switch valueType {
+				case telegraf.Counter:
+					acc.AddCounter(measurement,
+						map[string]interface{}{name: metric},
+						map[string]string{"type": s.sockType, "id": s.sockID, "collection": tag})
+				default:
+					acc.AddGauge(measurement, map[string]interface{}{name: metric}, map[string]string{"type": s.sockType, "id": s.sockID, "collection": tag})
+				}
```

Comment:
In case the lookup above fails, does it make sense to output those metrics?

Reply:
I believe so for backward compatibility reasons. I added logic to fall back into the previous logic of AddFields